File: javascript/point.js
var Point = function(x, y) {
    if (!(this instanceof Point)) {
        return new Point(x, y);
    }

    this.x = x;
    this.y = y;
};

Point.prototype = (function() {
    var distance = function() {
        return Math.sqrt(this.x * this.x + this.y * this.y);
    };

    return {
        distance: distance
    };
}());

exports.Point = Point;

File: go/point.go
package main

import "math"

type Point struct {
	x float64
	y float64
}

func newPoint(X float64, Y float64) Point {
	return Point{x: X, y: Y}
}

func (point Point) distance() float64 {
	return math.Sqrt(point.x*point.x + point.y*point.y)
}
